config/src/credentials.rs

```diff
@@ -124,17 +123,16 @@ struct KeyCloak {
 
 #[async_trait]
 impl Cred for KeyCloak {
-    fn get_request_metadata(&self) -> String {
+    async fn get_request_metadata(&self) -> String {
         if self.is_expired() {
-            self.refresh_rpt_token();
+            self.refresh_rpt_token().await;
```

Comment:
A token refresh will be initiated only when there are about 5 seconds for the token to expire. But this method is not invoked automatically every time a request is made. Shouldn't it be the case?